clients/cache/redis.go

```diff
@@ -0,0 +1,76 @@
+package cache
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/kava-labs/kava-proxy-service/logging"
+	"github.com/redis/go-redis/v9"
+)
+
+// RedisCache is an implementation of Cache that uses Redis as the caching backend.
+type RedisCache struct {
+	client *redis.Client
+	logger *logging.ServiceLogger
+}
+
+var _ Cache = (*RedisCache)(nil)
+
+func NewRedisCache(
+	ctx context.Context,
+	logger *logging.ServiceLogger,
```

Comment:
pass all this config in a single config struct, and don't pass the context in when initializing the cache, but when any of the methods such as set/get/delete are called

Reply:
make sense

Reply:
done